### PR TITLE
Add alert labels

### PR DESCRIPTION
**Pull request by MatanYadaev**
https://github.com/ballerine-io/ballerine/assets/13586343/7b2a38d2-c494-403f-b3d4-c19cc1a9694b

When the text is too long:
CleanShot 2024-03-12 at 12 56 28

### CODE REVIEW

File: services/workflows-service/prisma/migrations/20240311133449_add_label_to_alert_definitions/migration.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE "AlertDefinition" ADD COLUMN "label" TEXT NOT NULL DEFAULT substr(md5(random()::text), 0, 10);
```
Comment (MatanYadaev) [inline services/workflows-service/prisma/migrations/20240311133449_add_label_to_alert_definitions/migration.sql:1]: Added a default to not break local environments that already have alert definitions

File: apps/backoffice-v2/src/domains/alerts/query-keys.ts [hunk 2]
```diff
@@ -35,4 +39,8 @@ export const alertsQueryKeys = createQueryKeys('alerts', {
       queryFn: () => fetchAlertDefinitionByAlertId({ alertId }),
     };
   },
+  alertLabels: () => ({
+    queryKey: ['alertLabels'],
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/domains/alerts/query-keys.ts]: Pretty sure the key of the method in this case `alertLabels` is already applied by the library.
```suggestion
    queryKey: [{}],
```

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx [hunk 3]
```diff
@@ -44,6 +44,10 @@ export const columns = [
     },
     header: 'Date & Time',
   }),
+  columnHelper.accessor('label', {
+    cell: info => <Badge variant={'secondary'}>{info.getValue()}</Badge>,
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx]: Consider if `valueOrNA` or the component wrapper is needed here. Can label be falsy? And how does it look like when invalid data or no data returns from the API.

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/hooks/useTransactionMonitoringAlertsLogic/useTransactionMonitoringAlertsLogic.tsx [hunk 4]
```diff
@@ -29,6 +30,8 @@ export const useTransactionMonitoringAlertsLogic = () => {
         ?.sort((a, b) => (a?.id === session?.user?.id ? -1 : b?.id === session?.user?.id ? 1 : 0)),
     [assignees, session?.user?.id],
   );
+  const { data: labels } = useAlertLabelsQuery();
+  const sortedLabels = useMemo(() => labels?.sort(), [labels]);
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/hooks/useTransactionMonitoringAlertsLogic/useTransactionMonitoringAlertsLogic.tsx]: `Array.prototype.sort` mutates.
```suggestion
  const sortedLabels = useMemo(() => labels?.slice()?.sort(), [labels]);
```

Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/hooks/useTransactionMonitoringAlertsLogic/useTransactionMonitoringAlertsLogic.tsx]: Also is there a need for the labels to be sorted?

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx (hunk 3 above)
Comment (MatanYadaev) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx]: `label` is always a string

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/hooks/useTransactionMonitoringAlertsLogic/useTransactionMonitoringAlertsLogic.tsx (hunk 4 above)
Comment (MatanYadaev) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/hooks/useTransactionMonitoringAlertsLogic/useTransactionMonitoringAlertsLogic.tsx]: There's no clear reason for sorting the labels. Just kept the logic we have with `assignees`, and it's kinda makes sense to sort the label, but not important.

File: services/workflows-service/scripts/alerts/generate-alerts.ts [hunk 5]
```diff
@@ -133,7 +137,39 @@ export const ALERT_INLINE_RULES = [
       } as TransactionsAgainstDynamicRulesType,
     },
   },
-] as const satisfies readonly { inlineRule: InlineRule; defaultSeverity: AlertSeverity }[];
+] as const satisfies readonly {
+  inlineRule: InlineRule;
+  label: string;
+  defaultSeverity: AlertSeverity;
+}[];
```
Comment (Omri-Levy) [inline services/workflows-service/scripts/alerts/generate-alerts.ts]: `ReadonlyArray` might be more readable.

File: services/workflows-service/scripts/alerts/generate-alerts.ts [hunk 6]
```diff
@@ -133,7 +137,39 @@ export const ALERT_INLINE_RULES = [
...
+
+function createData(
```
Comment (Omri-Levy) [inline services/workflows-service/scripts/alerts/generate-alerts.ts]: Does ESLint not error on this function in the IDE?

Comment (MatanYadaev) [reply services/workflows-service/scripts/alerts/generate-alerts.ts]: No error on my machine. But I'll change it to an arrow function :)

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx [hunk 7]
```diff
@@ -44,6 +44,14 @@ export const columns = [
     },
     header: 'Date & Time',
   }),
+  columnHelper.accessor('label', {
+    cell: info => (
+      <Badge variant="secondary" className="max-w-[8rem]" title={info.getValue()}>
+        <div className="truncate">{info.getValue()}</div>
+      </Badge>
+    ),
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx]:
```suggestion
    cell: info => {
    const label = info.getValue();
    
    return (
      <Badge variant="secondary" className="max-w-[8rem]" title={label}>
        <div className="truncate">{label}</div>
      </Badge>
    )
    },
```

Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx]: Can't the `truncate` class work on the badge directly?

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/hooks/useTransactionMonitoringAlertsLogic/useTransactionMonitoringAlertsLogic.tsx (hunk 4 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/hooks/useTransactionMonitoringAlertsLogic/useTransactionMonitoringAlertsLogic.tsx]: Note the assignees is sorted so the signed in user is always at the top.

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx (hunk 7 above)
Comment (MatanYadaev) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx]: When the `truncate` is on the `Badge` component it doesn't look good, it ignores the padding, etc.

**Review by Omri-Levy — APPROVED** [review]

**Pull request merged**